deploy/tools/envs-validator/schema.ts

```diff
@@ -0,0 +1,339 @@
+    //   a. homepage
+    NEXT_PUBLIC_HOMEPAGE_CHARTS: yup
+      .array()
+      .transform(getEnvValue)
+      .json()
+      .of(yup.string<ChainIndicatorId>().oneOf([ 'daily_txs', 'coin_price', 'market_cap' ])),
+    NEXT_PUBLIC_HOMEPAGE_PLATE_TEXT_COLOR: yup.string(),
+    NEXT_PUBLIC_HOMEPAGE_PLATE_BACKGROUND: yup.string(),
+    NEXT_PUBLIC_HOMEPAGE_SHOW_GAS_TRACKER: yup.boolean(),
+    NEXT_PUBLIC_HOMEPAGE_SHOW_AVG_BLOCK_TIME: yup.boolean(),
+
+    //     b. sidebar
+    NEXT_PUBLIC_FEATURED_NETWORKS: yup
+      .array()
+      .json()
+      .of(featuredNetworkSchema),
+    NEXT_PUBLIC_OTHER_LINKS: yup
+      .array()
+      .transform(getEnvValue)
+      .json()
+      .of(navItemExternalSchema),
+    NEXT_PUBLIC_NETWORK_LOGO: yup.string().url(),
+    NEXT_PUBLIC_NETWORK_LOGO_DARK: yup.string().url(),
+    NEXT_PUBLIC_NETWORK_ICON: yup.string().url(),
+    NEXT_PUBLIC_NETWORK_ICON_DARK: yup.string().url(),
+
+    //     c. footer
+    NEXT_PUBLIC_FOOTER_LINKS: yup
+      .array()
+      .json()
+      .of(footerLinkGroupSchema),
+
+    //     d. views
+    NEXT_PUBLIC_VIEWS_BLOCK_HIDDEN_FIELDS: yup
+      .array()
+      .transform(getEnvValue)
+      .json()
+      .of(yup.string<BlockFieldId>().oneOf(BLOCK_FIELDS_IDS)),
+
+    //     e. misc
+    NEXT_PUBLIC_NETWORK_EXPLORERS: yup
+      .array()
+      .transform(getEnvValue)
+      .json()
+      .of(networkExplorerSchema),
+    NEXT_PUBLIC_HIDE_INDEXING_ALERT: yup.boolean(),
+
+    // 5. Features configuration
+    NEXT_PUBLIC_API_SPEC_URL: yup.string().url(),
+    NEXT_PUBLIC_STATS_API_HOST: yup.string().url(),
+    NEXT_PUBLIC_VISUALIZE_API_HOST: yup.string().url(),
+    NEXT_PUBLIC_CONTRACT_INFO_API_HOST: yup.string().url(),
+    NEXT_PUBLIC_GRAPHIQL_TRANSACTION: yup.string().matches(regexp.HEX_REGEXP),
+    NEXT_PUBLIC_WEB3_WALLETS: yup
+      .mixed()
+      .test('shape', 'Invalid schema were provided for NEXT_PUBLIC_WEB3_WALLETS, it should be either array or "none" string literal', (data) => {
+        const isNoneSchema = yup.string().equals([ 'none' ]);
+        const isArrayOfWalletsSchema = yup
+          .array()
+          .transform(getEnvValue)
+          .json()
+          .of(yup.string<WalletType>().oneOf(SUPPORTED_WALLETS));
+
+        return isNoneSchema.isValidSync(data) || isArrayOfWalletsSchema.isValidSync(data);
+      }),
+    NEXT_PUBLIC_WEB3_DISABLE_ADD_TOKEN_TO_WALLET: yup.boolean(),
+    NEXT_PUBLIC_AD_TEXT_PROVIDER: yup.string<AdTextProviders>().oneOf(SUPPORTED_AD_TEXT_PROVIDERS),
+
+    // 6. External services envs
+    NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID: yup.string(),
+    NEXT_PUBLIC_RE_CAPTCHA_APP_SITE_KEY: yup.string(),
+    NEXT_PUBLIC_GOOGLE_ANALYTICS_PROPERTY_ID: yup.string(),
+    NEXT_PUBLIC_MIXPANEL_PROJECT_TOKEN: yup.string(),
+  })
+  .concat(accountSchema)
+  .concat(adsBannerSchema)
+  .concat(marketplaceSchema)
+  .concat(rollupSchema)
+  .concat(beaconChainSchema)
+  .concat(sentrySchema);
+
+export default schema;
```

Comment:
😢

Reply:
I can split it into separate files, if it seems more readable for you